src/containerapp/azext_containerapp/containerapp_decorator.py

```diff
@@ -698,6 +708,47 @@ def set_up_system_assigned_identity_as_default_if_using_acr(self):
                         return
                 self.set_argument_registry_identity('system')
 
+    def set_up_consumption_gpu_wp_payload(self, consumption_gpu_profile_type):
```

Comment:
>set_up_consumption_gpu_wp_payload [](http://example.com/codeflow?start=8&length=33)

In validate_consumption_gpu_profile you also allow A10, but here you don't have that. Both A10 and NC12-A100 are not skus we support today, but I'm guessing you are future-proofing here.